ci-operator/step-registry/baremetalds/sno/hostname/baremetalds-sno-hostname-commands.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Fetch packet basic configuration
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/packet-conf.sh"
+
+ssh "${SSHOPTS[@]}" "root@${IP}" bash - << "EOF"
+
+hostname_script=$(cat << "IEOF"
+#!/usr/bin/env bash
+
+set -euoE pipefail
+
+export KUBECONFIG=/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/node-kubeconfigs/localhost.kubeconfig
+wait_for_api() {
+  echo "Waiting for API..."
+  until oc get clusterversion &> /dev/null
+  do
+    echo "Waiting for API..."
+    sleep 5
+  done
+  echo "API is available"
+}
+
+wait_approve_csr() {
+  local name=${1}
+
+  echo "Waiting for ${name} CSR..."
+  until oc get csr | grep -i "${name}" | grep -i "pending" &> /dev/null
+  do
+    echo "Waiting for ${name} CSR..."
+    sleep 5
+  done
+  echo "CSR ${name} is ready for approval"
+
+  echo "Approving all pending CSRs..."
+  oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' | xargs oc adm certificate approve
+}
+
+if [[ "$(hostname)" != "another-hostname" ]]
```

Comment:
There are some instances where RHCOS and networking will set a localhost hostname. I believe this is when DHCP assignments are slow; they would know best.

Reply:
This script runs as a [oneshot systemd service](https://github.com/openshift/release/pull/43483/files#diff-d608192592eb9f6facc80e13deb86ad1f90eb8f2eccafa971d76ecbf983cd37dR97-R106) after the `kubelet.service`, which in turn runs after the `network-online.target`. The latter requires the `node-valid-hostname.service`, which waits for `5min` for a non-localhost hostname (via executing `/usr/local/bin/mco-hostname --wait`).

IIUC what you are suggesting is that the `node-valid-hostname.service` will fail (e.g. because of a very slow DHCP assignment) and the node will still move forward with starting `kubelet` with a localhost hostname?